### PR TITLE
Asset browser pagination size changer

### LINKED ISSUE #3142: Asset Browser only allows to grab 30 files at once and not all of the folder.
## Bug Description
If you want to grab more than 30 files at once, you have to grab them one by one.

## How to Reproduce
Try to add for example 60 files at once.

## Environment
Statamic 3.0.38 Pro
Laravel 8.19.0
PHP 7.4.13

**Install method (choose one):**
- Fresh install from `statamic/statamic`

### PR DESCRIPTION
**Pull request by jackmcdade**
Uses localStorage to store the option so it'll persist for a while.

Between this feature and shift-to-select-multiple, should be able to resolve #3142.

### CODE REVIEW

File: src/Http/Controllers/CP/Assets/BrowserController.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function search(Request $request, $container)
             ? $container->searchIndex()->ensureExists()->search($request->search)
             : $container->queryAssets()->where('path', 'like', '%'.$request->search.'%');
 
-        $assets = $query->paginate(30);
+        $assets = $query->paginate(request('perPage', config('statamic.cp.pagination_size')));
```
Comment (jasonvarga) [inline src/Http/Controllers/CP/Assets/BrowserController.php:93]: I think `request('perPage')` is probably enough without the fallback. Pretty sure it always gets sent up. (We do it that way in the entry listing)

File: src/Http/Controllers/CP/Assets/BrowserController.php [hunk 2]
```diff
@@ -77,7 +77,7 @@ public function folder(Request $request, $container, $path = '/')
             $query->orderBy($request->sort, $request->order ?? 'asc');
         }
 
-        $assets = $query->paginate(30);
+        $assets = $query->paginate(request('perPage', config('statamic.cp.pagination_size')));
```
Comment (jasonvarga) [inline src/Http/Controllers/CP/Assets/BrowserController.php:80]: See other comment about `perPage`

**Review by jasonvarga — CHANGES_REQUESTED** [review]
We have a `HasPagination` mixin that the entries listing (Listing.vue) uses that already handles saving the pagination size to user preferences. Maybe we can use that instead?

**Comment by jackmcdade** [comment]
> We have a `HasPagination` mixin that the entries listing (Listing.vue) uses that already handles saving the pagination size to user preferences. Maybe we can use that instead?

Wouldn't that use the same `per_page` setting between entries and assets then though?

**Comment by jasonvarga** [comment]
It uses the `preferencesPrefix` property to save into different keys.

Here's how it does it in the entries/Listing component:
https://github.com/statamic/cms/blob/44cc9ea1cd234ed35164aa6404d5383fa290b8ef/resources/js/components/entries/Listing.vue#L122

You'd do something like `assets.${this.container.id}`

**Comment by jackmcdade** [comment]
Ah okay, I'll give it a shot.

**Review by jasonvarga — APPROVED** [review]
🔥

**Pull request merged**